components/patient-data/rest/src/test/java/org/phenotips/data/rest/internal/DefaultDomainObjectFactoryTest.java

```diff
@@ -0,0 +1,260 @@
+    }
+
+    @Test
+    public void createPatientPerformsCorrectly() throws ComponentLookupException, URISyntaxException, Exception
+    {
+        XWikiDocument documentReference = mock(XWikiDocument.class);
+        DocumentReference patientDocument = mock(DocumentReference.class);
+        DocumentReference creatorDocument = mock(DocumentReference.class);
+        UriBuilder uriBuilder = mock(UriBuilder.class);
+        URI uri = new URI("uri");
+        Date creationDate = new Date(0);
+        DateTime creationDateTime = new DateTime(creationDate).withZone(DateTimeZone.UTC);
+
+        when(this.patient.getDocument()).thenReturn(patientDocument);
+        when(this.access.hasAccess(Right.VIEW, null, patientDocument)).thenReturn(true);
+        when(this.documentAccessBridge.getDocument(patientDocument)).thenReturn(documentReference);
+        when(this.patient.getId()).thenReturn("id");
+        when(this.patient.getExternalId()).thenReturn("externalid");
+        when(this.patient.getReporter()).thenReturn(creatorDocument);
+
+        when(creatorDocument.toString()).thenReturn("creator");
+        when(documentReference.getAuthorReference()).thenReturn(creatorDocument);
+        when(documentReference.getVersion()).thenReturn("version");
+        when(documentReference.getCreationDate()).thenReturn(creationDate);
+        when(documentReference.getDate()).thenReturn(creationDate);
+        when(this.uriInfo.getBaseUriBuilder()).thenReturn(uriBuilder);
+        when(uriBuilder.path(PatientResource.class)).thenReturn(uriBuilder);
+        when(uriBuilder.build("id")).thenReturn(uri);
+
+        PatientSummary patientSummary = mocker.getComponentUnderTest().createPatientSummary(this.patient, this.uriInfo);
+
+        assertEquals("id", patientSummary.getId());
+        assertEquals("externalid", patientSummary.getEid());
+        assertEquals("creator", patientSummary.getCreatedBy());
+        assertEquals("creator", patientSummary.getLastModifiedBy());
+        assertEquals("version", patientSummary.getVersion());
+        assertEquals(creationDateTime, patientSummary.getCreatedOn());
+        assertEquals(creationDateTime, patientSummary.getLastModifiedOn());
+        assertEquals(1, patientSummary.getLinks().size());
+        assertEquals("uri", patientSummary.getLinks().get(0).getHref());
+    }
+
+    @Test
+    public void createPatientFromSummaryWrongLength() throws Exception
+    {
+        Object[] summary = {new Object()};
+        assertNull(this.mocker.getComponentUnderTest().createPatientSummary(summary, this.uriInfo));
+    }
+
+    @Test
+    public void createPatientFromSummaryWrongObjectTypes() throws Exception
+    {
+        Object[] summary = new Object[7];
+        assertNull(this.mocker.getComponentUnderTest().createPatientSummary(summary, this.uriInfo));
+    }
+
+    @Test
+    public void createPatientFromSummaryNoAccess() throws Exception
+    {
+        Object[] summary = {new Object()};
```

Comment:
`Object[] summary = new Object[7]` the null return value you are getting currently is due to the `summaryData.length != 7` condition failing